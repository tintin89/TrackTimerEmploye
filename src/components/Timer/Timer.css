
.timer__container{
    width: 90%;
    max-width: 500px;
    display: flex;
    flex-flow: column;
    align-items: center;
}
.timer{
   color: #ffff;
   width: 100%;
   margin-top: 2rem;
   display: flex;
   justify-content: center;
   font-size: 2rem;
   font-family: 'Poppins', sans-serif;   
}
.timer__line{
    height: 1px;
    background-color: #ffff;
    width: 30%;
}

.expectedHours{
  margin-top: 0.5rem;
  font-size: 1rem;
  font-family: 'Poppins', sans-serif;
  color: #ffff;
  text-align: center;
}

.lunchTime{
  background-color: #1c1f24;
  margin-top: 3rem;
  display: flex;
  justify-content: center;
  width: 90%;
  max-width: 200px;
  padding: 0.5rem;
  color: #ffff;
  border-radius: 3px;
  font-family: 'Poppins', sans-serif;
  box-shadow: 1px 1px 3px rgba(0, 0, 0, 0.5);
  cursor: pointer;
}