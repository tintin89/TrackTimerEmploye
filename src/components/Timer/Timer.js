import React,{useState,useEffect} from 'react';
import './Timer.css'
import axios from 'axios';
import {useSelector} from 'react-redux';
import moment from 'moment'


const mapState = (globalState)=>({
    userInfo:globalState.userInfo,
    userData:globalState.userData
})


function Timer() {  
    const {userInfo,userData} = useSelector(mapState);  
    const [isLunching,setisLunching] = useState(localStorage.getItem('isLunching') ? JSON.parse(localStorage.getItem('isLunching')) : false);  
    const [employeTimer,setEmployeTimer] = useState(localStorage.getItem('employeTimer') ? JSON.parse(localStorage.getItem('employeTimer')) : {
        second:'00',
        minute:'00',
        hours:'00',
        counter:0
    });

    
  

    useEffect(()=>{
      let intervalId;
       if(!isLunching){      
        if(localStorage.getItem("existCounter")){
            localStorage.removeItem("existCounter");
            setEmployeTimer({...employeTimer,counter:userData})
        }

        intervalId = setInterval(()=>{
          
            const hours = Math.floor((employeTimer.counter/(60*60)));
            const minutes = Math.floor((employeTimer.counter/60)%60);
            const seconds = employeTimer.counter%60;
            const formatedHours = String(hours).length === 1 ? `0${hours}` : hours;
            const formatedSecond = String(seconds).length === 1 ? `0${seconds}`: seconds;
            const formatedMinute = String(minutes).length === 1 ? `0${minutes}`: minutes;
           
           setEmployeTimer(prevState=>({
                second:formatedSecond,
                minute:formatedMinute,
                hours:formatedHours,
                counter:prevState.counter +1
            }));
            localStorage.setItem("employeTimer",JSON.stringify(employeTimer));
        
        },1000)      
       }
       return ()=>{
            
          return clearInterval(intervalId)
        };

    },[employeTimer,isLunching,userData])



    const handleLunch = async () =>{    
        if(isLunching){    
        
       try {
             await axios.put(`https://timertracker-a17c6-default-rtdb.firebaseio.com/employers/${userInfo.localId}/${moment().format("MMM Do YY")}/lunchBreaks/${localStorage.getItem("lunchBStart")}.json`,{
               finishLunch:moment().format('LTS')
               })                 
          } catch (error) {
              console.log(error)
          }     

         setisLunching(false);
         localStorage.setItem("isLunching",false);  
         localStorage.removeItem("lunchBStart");     
        }else{
            setisLunching(true);
            localStorage.setItem("isLunching",true);
            localStorage.setItem("lunchBStart",moment().format('LTS'))
        } 
         
        }

    return (
         <>
       <div className="timer__container">
            <div className="timer">
            <span>{employeTimer.hours}:</span>
            <span>{employeTimer.minute}:</span>
            <span>{employeTimer.second}</span>
            </div>
            <div className="timer__line"></div>
            <span className="expectedHours">Expected Working Hours: 8hrs</span>            
        </div>
         <div onClick={()=>handleLunch()} className="lunchTime">
         {isLunching ? "Finish Lunch" : "Go to Lunch"}
     </div>
         </>
    )
}

export default Timer
