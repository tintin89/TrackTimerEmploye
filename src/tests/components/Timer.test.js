import Timer,{handleLunch} from '../../components/Timer/Timer'
import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);



describe('test in Employes.js',()=>{
    const store = mockStore({
        userinfo:"",
        userData:"",
        
    });
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
              <Timer/>  
            </MemoryRouter>
            
        </Provider> )

    test('Must show correctly',()=>{
     
        expect(wrapper).toMatchSnapshot();
    })

   
})