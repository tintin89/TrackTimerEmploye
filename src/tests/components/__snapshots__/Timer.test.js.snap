// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in Employes.js Must show correctly 1`] = `
Array [
  <div
    className="timer__container"
  >
    <div
      className="timer"
    >
      <span>
        00
        :
      </span>
      <span>
        00
        :
      </span>
      <span>
        00
      </span>
    </div>
    <div
      className="timer__line"
    />
    <span
      className="expectedHours"
    >
      Expected Working Hours: 8hrs
    </span>
  </div>,
  <div
    className="lunchTime"
    onClick={[Function]}
  >
    Go to Lunch
  </div>,
]
`;
